queries.sql

```diff
@@ -22,6 +22,61 @@ select * from animals where name != 'Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 select * from animals where weight_kg between 10.4 and 17.3;
 
+-- Set the species column to unspecified inside a transaction then roll back
+begin;
+update animals
+set species = 'unspecified' where species is null;
+rollback;
 
+-- start a transaction 1
+begin;
+-- Update the animals table by setting the species column to 'digimon' for all animals that have a name ending in 'mon'
+update animals
+set species = 'digimon' where name like '%mon';
 
+-- Update the animals table by setting the species column to 'pokemon' for all animals that don't have species already set.
+update animals
+set species = 'pokemon' where species is null;
+-- Commit the trabsaction
+commit;
 
+-- start a transaction 2
+begin;
+-- delete all records in the animals
+delete from animals;
+-- roll back the transaction
+rollback;
+
+-- start a transaction 3
+begin;
+-- Delete all animals born after Jan 1st, 2022
+delete from animals where date_of_birth > '2022-01-01';
+-- Create a savepoint for the transaction.
+savepoint sp1;
+-- Update all animals' weight to be their weight multiplied by -1.
+update animals
+set weight_kg = -1 * weight_kg;
+-- Rollback to the savepoint
+rollback to sp1;
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+update animals
+set weight_kg = -1 * weight_kg where weight_kg < 0;
+-- Commit transaction
+commit;
+
+-- Write queries to answer the following questions:
+-- How many animals are there?
+select count(id) from animals;
+-- How many animals have never tried to escape?
+select count(escape_attempts) from animals where escape_attempts = 0;
+-- What is the average weight of animals?
+select avg(weight_kg) from animals;
+-- Who escapes the most, neutered or not neutered animals?
+select neutered, max(escape_attempts) from animals
+group by neutered;
+-- What is the minimum and maximum weight of each type of animal?
+select species, min(weight_kg), max(weight_kg) from animals
+group by species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+select species, avg(escape_attempts) from animals where date_of_birth between '1990-01-01' and '2000-12-31'
+group by species;
```

Comment:
Hello, I'm here to review your code. You have done a partial job on this project.

One thing I noticed is that you have a syntax error in your WHERE clause. You placed the WHERE clause after the GROUP BY clause, which is not allowed in SQL. The WHERE clause should always come before the GROUP BY clause, as it filters the rows before they are grouped.

Here's an example of how you can write your code that meets all the project requirements:

```sql
-- Select species and average number of escape attempts from animals
SELECT species, AVG(escape_attempts) FROM animals
-- Filter rows by date of birth between 1990 and 2000
WHERE date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31'
-- Group rows by species
GROUP BY species;

Reply:
Hi @topeogunleye , I really appreciate the time and effort you invested in reviewing my project.  in the screenshot that you provided, the `where` clause is on line 81 while `group by` is on line 82. So I guess there is not any error in the code.

Thank you for your feedback.

Reply:
Hi, @momo-87,

You’re very welcome. I’m glad to hear that you made the necessary changes to your project. You have done a great job of following the instructions and applying the SQL concepts. Your code is well-written, clear, and efficient.